File: week-7/assignment-7/go-ride-sharing-system/pkg/fileutils/fileutils.go
package fileutils

import (
	"bufio"
	"os"

	"rideshare/pkg/logger"
)

/*
	Writing results to file with error handling
*/

func WriteResultsToFile(results []string, fileName string) {
	// Ensure outputs directory exists
	os.MkdirAll("outputs", os.ModePerm)

	file, err := os.OpenFile("outputs/"+fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		logger.Logger.Println("Failed to open file:", err)
		return
	}
	defer file.Close()

	writer := bufio.NewWriter(file)
	for _, line := range results {
		_, err := writer.WriteString(line + "\n")
		if err != nil {
			logger.Logger.Println("Failed to write result:", err)
		}
	}
	writer.Flush()
}

File: week-7/assignment-7/java-ride-sharing-system/src/com/rideshare/FileUtils.java
package com.rideshare;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.List;
import java.util.logging.Logger;

/**
 * Utility for writing results to a file.
 * - Logs errors during file operations.
 */
public class FileUtils {
    private static final String OUTPUT_DIR = "outputs";
    private static final String RESULT_FILE = OUTPUT_DIR + "/ride_results.txt";

    public static void writeResultsToFile(List<String> results, String fileName) {
        // Use try-with-resources to ensure file is closed
        try {
            // Ensure outputs directory exists
            new File(OUTPUT_DIR).mkdirs();

            BufferedWriter writer = new BufferedWriter(new FileWriter(RESULT_FILE, true));
            for (String line : results) {
                writer.write(line);
                writer.newLine();
            }
            writer.close();
        } catch (IOException e) {
            // Handle IOException
            LoggerConfig.LOGGER.severe("Failed to write results to file: " + e.getMessage());
        }
    }
}
